packages/grafana-ui/src/components/Modal/Modal.tsx

```diff
@@ -19,48 +19,51 @@ export interface Props extends Themeable {
 
   /** If not set will call onDismiss if that is set. */
   onClickBackdrop?: () => void;
+  render?: (props: RenderApi) => JSX.Element;
 }
 
-export class UnthemedModal extends React.PureComponent<Props> {
-  onDismiss = () => {
-    if (this.props.onDismiss) {
-      this.props.onDismiss();
-    }
-  };
-
-  onClickBackdrop = () => {
-    this.onDismiss();
-  };
+interface RenderApi {
+  styles: ReturnType<typeof getModalStyles>;
+}
 
-  renderDefaultHeader(title: string) {
-    const { icon, iconTooltip } = this.props;
+export function Modal(props: PropsWithChildren<Props>): JSX.Element | null {
+  const { title, children, isOpen = false, className, onDismiss: propsOnDismiss, onClickBackdrop, render } = props;
+  const theme = useTheme();
+  const styles = getModalStyles(theme);
+  const onDismiss = useCallback(() => {
+    if (propsOnDismiss) {
+      propsOnDismiss();
+    }
+  }, [propsOnDismiss]);
 
-    return <ModalHeader icon={icon} iconTooltip={iconTooltip} title={title} />;
+  if (!isOpen) {
+    return null;
   }
 
-  render() {
-    const { title, isOpen = false, theme, className } = this.props;
-    const styles = getModalStyles(theme);
-
-    if (!isOpen) {
-      return null;
-    }
-
-    return (
-      <Portal>
-        <div className={cx(styles.modal, className)}>
-          <div className={styles.modalHeader}>
-            {typeof title === 'string' ? this.renderDefaultHeader(title) : title}
-            <div className={styles.modalHeaderClose}>
-              <IconButton surface="header" name="times" size="lg" onClick={this.onDismiss} />
-            </div>
+  return (
+    <Portal>
+      <div className={cx(styles.modal, className)}>
+        <div className={styles.modalHeader}>
+          {typeof title === 'string' && <DefaultModalHeader {...props} title={title} />}
+          {typeof title !== 'string' && title}
+          <div className={styles.modalHeaderClose}>
+            <IconButton surface="header" name="times" size="lg" onClick={onDismiss} />
           </div>
-          <div className={styles.modalContent}>{this.props.children}</div>
         </div>
-        <div className={styles.modalBackdrop} onClick={this.props.onClickBackdrop || this.onClickBackdrop} />
-      </Portal>
-    );
-  }
+        {render && render({ styles })}
```

Comment:
Is there a specific reason we'd need `render` in addition to `children` prop? Seems like they both are used to render content. If we need to customize content container styling maybe introducing `contentClassName` prop would work instead?

Reply:
The reason for this is that I didn't want to break any existing Modals.